File: Backend/dgulaf/src/main/java/com/dgulaf/service/FoundItemService.java
package com.dgulaf.service;

import com.dgulaf.model.FoundItem;
import com.dgulaf.repository.FoundItemRepository;
import org.springframework.stereotype.Service;

@Service
public class FoundItemService {
    private final FoundItemRepository foundItemRepository;

    public FoundItemService(FoundItemRepository foundItemRepository) {
        this.foundItemRepository = foundItemRepository;
    }

    // createFoundItem으로 이름을 변경하고, FoundItem을 저장하는 메서드로 수정
    public FoundItem createFoundItem(FoundItem foundItem) {
        return foundItemRepository.save(foundItem);  // FoundItem을 저장
    }

    // ID로 FoundItem을 조회하는 메서드
    public FoundItem getFoundItemById(int id) {
        return foundItemRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("FoundItem not found with id: " + id));
    }
}

File: Frontend/dgu_laf/lib/service/found_item_service.dart
import 'dart:convert';
import 'package:dgu_laf/model/found_item.dart';
import 'package:http/http.dart' as http;

class FoundItemService {
  static const String baseUrl = 'http://192.168.219.105:8080/found-items';

  // CREATE
  static Future<FoundItem> createFoundItem(FoundItem foundItem) async {
    final response = await http.post(
      Uri.parse(baseUrl),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(foundItem.toJson()),
    );

    if (response.statusCode == 200) {
      return FoundItem.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to create found item');
    }
  }

  // READ
  static Future<FoundItem> getFoundItemById(int id) async {
    final response = await http.get(Uri.parse('$baseUrl/$id'));

    if (response.statusCode == 200) {
      return FoundItem.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to load found item');
    }
  }

  // UPDATE
  static Future<FoundItem> updateFoundItem(int id, FoundItem foundItem) async {
    final response = await http.put(
      Uri.parse('$baseUrl/$id'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(foundItem.toJson()),
    );

    if (response.statusCode == 200) {
      return FoundItem.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to update found item');
    }
  }

  // DELETE
  static Future<void> deleteFoundItem(int id) async {
    final response = await http.delete(Uri.parse('$baseUrl/$id'));

    if (response.statusCode != 204) {
      throw Exception('Failed to delete found item');
    }
  }
}
